src/FadingCounter.h

```diff
@@ -15,26 +15,28 @@
 class FadingCounter
 {
 public:
+    static const size_t Precision = 10; ///< #counting slots, controls measur. accuracy
+
     FadingCounter();
 
     /// 0=remember nothing; -1=forget nothing; new value triggers clear()
-    void configure(double horizonSeconds);
+    void configure(const time_t horizonSeconds);
```

Comment:
One step at a time. Using a `std::chrono` here requires `squid.conf` parser logic and modifications to the internal logic of `FadingCounter`. They are explicitly being left to followup PR.

Reply:
> Using a `std::chrono` here requires `squid.conf` parser logic 

I seriously doubt that using std::chrono for FadingCounter::horizon requires any squid.conf parser changes, but, if it really does, then FadingCounter::horizon type and similar changes are not compatible with an "Update FadingCounter API" PR (because they require too many out-of-scope changes).

> and modifications to the internal logic of `FadingCounter`.

I do not think so, but our definitions of "internal logic" may differ.

> They are explicitly being left to followup PR.

In summary, it is totally fine to postpone conversion of FadingCounter internals to std::chrono beyond this PR as long as this PR does not change FadingCounter::horizon type (or contains other similar internal FadingCounter changes that are difficult to do correctly, especially without std::chrono help).

Reply:
`FadingCounter::horizon` is set by several configuration options. Currently they use `parse_time()`. I do not want to extend scope of this specific PR outside of `FadingCounter`.